### DISCUSSION TITLE
Ruby binding

### DISCUSSION THREAD

**Discussion by chrisliaw** [open-quantum-safe/liboqs#1598] [General]
Hi to the amazing liboqs project team!

I've built the Ruby binding for the liboqs, is that something interested for the team to include in the supported language?

The repository is here : https://github.com/chrisliaw/liboqs-ruby

Thanks!

Regards

**Comment by baentsch** [comment]
Hi Chris -- thanks for your nice words and your work on this language wrapper. Do you plan to actively maintain this project or do you expect us to do that? A quick glance at the project tells me that it seems to be inactive and further documents defects since a long time (August 2021 is shown as last commit date): Is this understanding correct? Does it work with the most recent `liboqs` release, for example?  Do you know of users for this package?

**Reply by chrisliaw** [reply]
Hi Michael!

Yes I have plan to maintain it as i will using it on another project using the oqs directly from Ruby. Will try to match with release together with liboqs releases.

But this liboqs-ruby project was started way before my this another project started so yeah It will pick up the pace. I think it was tested with 0.7.0 then.

Yes it is tested with liboqs commit 02da5677ea070c50d3907b8e6bff31a32e3cce14 on 30 Oct 2023 from the main. Next will sync with the tagged version when there is an release? 

Thanks!

**Reply by chrisliaw** [reply]
Hi @baentsch I think I've made a mistake in package naming now the package is oqs.gem which is not I intended to use.

Let me create another project with proper name and let you know?

Thanks!

**Comment by chrisliaw** [comment]
My new repos is at https://github.com/chrisliaw/roqs

See if Ruby binding is something the team is looking at?

Thanks!~

**Reply by dstebila** [reply]
Hi Chris! Great work!  Yes, I think we would be happy to have a Ruby binding included in our suite of language wrappers.  We don't have much expertise on Ruby within our team, so we'd be looking for you to join and continue to support and develop the sub-project.

**Reply by chrisliaw** [reply]
Great to know that!

Please do let me know if you need help on those area

Thanks!
